### PR TITLE
DOC: Review and update installation instructions.

### PR DESCRIPTION
**Pull request by JDWarner**
Documentation fixes after #1526

### CODE REVIEW

File: doc/source/install.txt [hunk 1]
```diff
@@ -42,40 +41,38 @@ them, scikit-image should already be installed.
...
 the `scipy section`_, `numpy section`_ and `scikit-image section`_. Make sure
-you download the right version for your system. E.g. numpy for Python 3.4 64
+you download the right version for your system. E.g., numpy for Python 3.4 64
 bit would be ``numpy‑1.9.2+mkl‑cp34‑none‑win_amd64.whl``.
 
 To install Goehlke's packages, use pip::
 
   pip install wheel
-  pip install --find-links Downloads scikit-image
+  pip install --find-links path/to/downloaded/packages scikit-image
```
Comment (arve0) [inline doc/source/install.txt]: Does forward slash work? Anyhow, backslash are probably more known for windows users.

File: doc/source/install.txt [hunk 2]
```diff
@@ -102,7 +99,7 @@ Or if you use Python 3::
 
 On Ubuntu scikit-image is found in the `universe repo`_, and python-skimage can
 also be found in the `Neurodebian repository`_. For using the repository
-follow the `Neurodebian instructions`_ to add Neurodebian to your system 
+follow the `Neurodebian instructions`_ to add Neurodebian to your system
```
Comment (arve0) [inline doc/source/install.txt]: I just remembered, the `Neurodebian instructions` link is outdated.

File: doc/source/install.txt (hunk 1 above)
Comment (JDWarner) [inline doc/source/install.txt]: Forward slash is understood in Python on all operating systems, to the best of my knowledge, and pip should parse it correctly.

**Comment by JDWarner** [comment]
Closed in favor of #1535

**Pull request closed without merging**